nexus/src/nexus.rs

```diff
@@ -142,8 +144,8 @@ impl Nexus {
      * Create a new Nexus instance for the given rack id `rack_id`
      */
     /* TODO-polish revisit rack metadata */
-    pub fn new_with_id(
-        rack_id: &Uuid,
+    pub async fn new_with_id(
```

Comment:
Wondering out loud: I've been trying to keep this sync because it's tempting to put async things in here that belong as background processes (e.g., saga recovery or database initialization).  I don't fully grok the flow of the root JSON file but it looks like this _is_ reasonable to be done immediately.  But I wonder if the (async) caller should pass in the file contents instead?

Reply:
We could also keep it sync and just use `std::fs::read`. Root roles are generally expected to be very small (hence why we're just keeping it in memory).

Alternatively we can just pass along the `PathBuf` of the root to Nexus, and Nexus can read it whenever `/updates/refresh` is hit.